src/export/StructureDefinitionExporter.ts

```diff
@@ -408,6 +409,16 @@ export class StructureDefinitionExporter implements Fishable {
     }
 
     const parentName = fshDefinition.parent || 'Resource';
+
+    if (fshDefinition.name === fshDefinition.parent) {
```

Comment:
I think on line 413 the check should be against `parentName` and not `fshDefinition.parent`. Currently I could do:
```
Profile: Resource
* rules
```
and still get the maximum call stack error. I do also get an error from the grammar, because the grammar wants parent to be there, but if we bother to do the `parentName` default setting on line 411 I think we should use it here.

Reply:
I think you're right that this should use `parentName`. I think I had originally switched back to using `fshDefinition.parent` because I thought it would help in the case when we export an Extension, but I don't think it will affect that case and we should support the case you mentioned above.